mariadb-database-id/3.initialize-database.sql

```diff
@@ -1,15 +1,16 @@
 USE eventuate;
 
 CREATE TABLE new_message (
-  id BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  id VARCHAR(767),
+  xid BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,
   destination LONGTEXT NOT NULL,
   headers LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
   payload LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
   published SMALLINT DEFAULT 0,
   creation_time BIGINT
 );
 
-INSERT INTO new_message (destination, headers, payload, published, creation_time) SELECT destination, headers, payload, published, creation_time FROM message;
+INSERT INTO new_message (id, destination, headers, payload, published, creation_time) SELECT "", destination, headers, payload, published, creation_time FROM message;
```

Comment:
@cer Chris,

> Why is "" used as the value for ID ? It should be MESSAGE.ID?

you also said:

> Perhaps the pk for existing events could just start at 0. I don't think the value matters (since the messages already have an ID header) 

So I thought, that having ID column is necessary only for compatibility, so I left it empty for migrated data.
But, I will fix it.

> Having said that, I don't think there is any point in have DB migration logic in the scripts that are used to build a Docker image since there is no data to migrate.

Actually I used it as smoke test that sql correct. And to show you how I see the migration script.
Since new id generation implementation is not yet finished, it is too early to work on complete migration scenarios and migration tests, I think. Also we probably need to discuss, where to store migration scripts, and how to test migration (I will create a separate thread).

Reply:
I don't think the value matters = the PK for existing events = auto-generated ID - just needs to be unique.

It's a bad idea to discard important data, especially when there is a column.

Reply:
ok, I will fix